openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -68,12 +70,19 @@ def get_handler_url(request, usage_key_str, handler_name):
     return Response({"handler_url": handler_url})
 
 
-@api_view(['GET', 'POST', 'PUT', 'PATCH', 'DELETE'])
-@authentication_classes([])  # Disable session authentication; we don't need it and don't want CSRF checks
-@permission_classes((permissions.AllowAny, ))
+# We cannot use DRF for this endpoint because its Request object is incompatible
+# with the API expected by XBlock handlers.
+# See https://github.com/edx/edx-platform/pull/19253
+# and https://github.com/edx/XBlock/pull/383 for context.
+@csrf_exempt
+@xframe_options_exempt
```

Comment:
> I'm not clear on this part–the DRF api-view decorator here was working but won't work with the LabX frontend? Or was this always a bit broken?

This was always a bit broken, but only certain types of POST request would trigger the broken behavior, and I guess I hadn't encountered any of those specific requests in my previous testing.

@symbolist actually [warned me about this in the initial code review](https://github.com/edx/edx-platform/pull/20645#discussion_r311298764) when adding this API, but I didn't fix it at the time because I wasn't seeing the issue. I guess I just wasn't testing with the right blocks / code paths.

This change is similar to a previous change already made for edx-platform's modulestore version of this same API.